### PR TITLE
optional description in modules/net-vpc-swp

### PR DESCRIPTION
**Pull request by skalolazka**
Part of https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/issues/1491

### CODE REVIEW

**Comment by skalolazka** [comment]
OK, so I will probably add more different variables for description to have more flexibility. So gimme a sec.

**Comment by ludoo** [comment]
Please don't add variables.

**Comment by ludoo** [comment]
What I mean a top-level optional description for the main resource is fine, all the rest should be optional attributes in existing object type variables.

File: modules/net-vpc-swp/variables.tf [hunk 1]
```diff
@@ -62,6 +74,7 @@ variable "policy_rules" {
       action                 = optional(string, "ALLOW")
       enabled                = optional(bool, true)
       tls_inspection_enabled = optional(bool, false)
+      description            = optional(string, "Managed by Terraform.")
```
Comment (ludoo) [inline modules/net-vpc-swp/variables.tf:77]: I would use `optional(string)` and then internally use the top-level `description` variable if this is null.

File: modules/net-vpc-swp/variables.tf [hunk 2]
```diff
@@ -35,6 +35,18 @@ variable "delete_swg_autogen_router_on_destroy" {
   default     = true
 }
 
+variable "description" {
+  description = "Optional description for the SWG."
+  type        = string
+  default     = "Managed by Terraform."
+}
+
+variable "gateway_security_policy_description" {
```
[the remark this answers is not in the document]
Comment (skalolazka) [reply modules/net-vpc-swp/variables.tf:44]: Should I make the "description" variable an object and add this there?

File: modules/net-vpc-swp/variables.tf (hunk 1 above)
Comment (skalolazka) [reply modules/net-vpc-swp/variables.tf:77]: Good point, can do.

File: modules/net-vpc-swp/variables.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (ludoo) [reply modules/net-vpc-swp/variables.tf:44]: If you do it your way, the new description needs to be specified everywhere which is a PITA.

[the remark this answers is not in the document]
Comment (skalolazka) [reply modules/net-vpc-swp/variables.tf:44]: OK, I see. Sure.

File: modules/net-vpc-swp/main.tf [hunk 3]
```diff
@@ -33,7 +33,7 @@ resource "google_network_security_tls_inspection_policy" "tls-policy" {
   project               = var.project_id
   name                  = var.name
   location              = var.region
-  description           = var.tls_inspection_config.description
+  description           = var.tls_inspection_config.description != null ? var.tls_inspection_config.description : var.description
```
Comment (ludoo) [inline modules/net-vpc-swp/main.tf:36]: you could do

`coalesce(var.tls_inspection_config.description , var.description)`

Comment (skalolazka) [reply modules/net-vpc-swp/main.tf]: Brilliant, thanks, added

File: modules/net-vpc-swp/variables.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (skalolazka) [reply modules/net-vpc-swp/variables.tf]: Should now be resolved, I believe

**Review by ludoo — APPROVED** [review]
Thanks!

**Pull request merged**